com.unity.uiwidgets/Runtime/engine2/UIWidgetsPanel.cs

```diff
@@ -145,25 +144,67 @@ public struct TextFont {
         }
         
         static bool _ShowDebugLog = false;
+        
+        protected virtual void loadConfiguration() {}
+
+        Dictionary<string,UIWidgetsPanel.TextFont> _internalTextFonts = new Dictionary<string, UIWidgetsPanel.TextFont>();
+        protected void AddFont(string family, List<string> assets, List<int> weights) {
+            if (assets.Count != weights.Count) {
+                UnityEngine.Debug.Log($"The size of {family}‘s assets should be equal to the weights'.");
+                return;
+            }
+            if (!_internalTextFonts.ContainsKey(family)) {
+                UIWidgetsPanel.TextFont textFont = new UIWidgetsPanel.TextFont();
+                textFont.family = family;
+                UIWidgetsPanel.Font[] fonts = new UIWidgetsPanel.Font[assets.Count];
+                for (int j = 0; j < assets.Count; j++) {
+                    UIWidgetsPanel.Font font = new UIWidgetsPanel.Font();
+                    font.asset = assets[j];
+                    font.weight = weights[j];
+                    fonts[j] = font;
+                }
+                textFont.fonts = fonts;
+                _internalTextFonts.Add(family,textFont);
+            }
+        }
 
         protected void OnEnable() {
             base.OnEnable();
+            D.assert(_wrapper == null);
+            _wrapper = new UIWidgetsPanelWrapper();
             var settings = new Dictionary<string, object>();
```

Comment:
we need refactor the code below as something like:

//load common configurations
loadConfiguation();

if (fonts != null && fonts.Length > 0) 
{
   foreach item in fonts: {
           AddFont(item.family, item.font, item.weights);
     }
}